### DISCUSSION TITLE
How to set the size of an egui::Window::new()

### DISCUSSION THREAD

**Discussion by jondavis847** [emilk/egui#4339] [Q&A]
```
use eframe::egui;

fn main() -> Result<(), eframe::Error> {
    env_logger::init(); // Log to stderr (if you run with `RUST_LOG=debug`).
    let options = eframe::NativeOptions {
        viewport: egui::ViewportBuilder::default().with_inner_size([1000.0, 1000.0]),
        ..Default::default()
    };
    eframe::run_native(
        "spacecraft builder",
        options,
        Box::new(|cc| {
            // This gives us image support:
            Box::<SpacecraftBuilder>::default()
        }),
    )
}

struct SpacecraftBuilder;

impl Default for SpacecraftBuilder {
    fn default() -> Self {
        Self {            
        }
    }
}

impl eframe::App for SpacecraftBuilder {
    fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {        

        let body_frame = egui::containers::Frame::none().fill(egui::Color32::from_gray(80));

        egui::Window::new("body_window")
            .default_width(500.0)
            .default_height(500.0)
            .frame(body_frame)
            .resizable(true)
            //.anchor(egui::Align2::LEFT_TOP, [0.0, 0.0])
            //.title_bar(false)
            .show(ctx, |ui| {
                ui.label("Hello World!");
                ui.button("testing")
            });
    }
}
```

Hello,

I'm just getting started with egui, basically just made it past the hello world example. I know that my gui eventually will have multiple windows that don't necessarily fit the default panel model of side, topbottom, central. I attempted just changing the hello world example from a central panel to a window. However, having tried every combo of default_width, fixed_size, min_size, etc. that I can in the API, the window always seems to be auto-sized to the content. I also cannot resize the window. I think I must be missing something.

![image](https://github.com/emilk/egui/assets/91633610/77835642-2c61-44c4-99f6-97c83d7661fe)

**Comment by thmxv** [comment]
If the window is meant to be smaller than all of its content, I think, that you need to put a [`ScrollArea`](https://docs.rs/egui/latest/egui/containers/scroll_area/struct.ScrollArea.html) inside the window and the content inside this `ScrollArea`.

**Comment by YgorSouza** [comment]
The demo app has some samples in the Window resize demo.

https://github.com/emilk/egui/blob/78d95f430b09e3040d3f383fde8fea2ae1592afb/crates/egui_demo_lib/src/demo/tests.rs#L581-L594
